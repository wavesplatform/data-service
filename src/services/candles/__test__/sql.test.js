const { interval } = require('../../../types/');
const { highestDividerLessThen, sql } = require('../sql');

describe('candles sql helper functions', () => {
  it('highest divider less then', () => {
    expect(
      highestDividerLessThen(interval('1m').unsafeGet(), ['1m', '1h', '1d'])
        .length
    ).toBe(interval('1m').unsafeGet().length);
    expect(
      highestDividerLessThen(interval('10m').unsafeGet(), ['5m', '15m', '1h'])
        .length
    ).toBe(interval('5m').unsafeGet().length);
    expect(
      highestDividerLessThen(interval('15m').unsafeGet(), ['5m', '15m', '1h'])
        .length
    ).toBe(interval('15m').unsafeGet().length);
    expect(
      highestDividerLessThen(interval('1h').unsafeGet(), ['1m', '1h', '1d'])
        .length
    ).toBe(interval('1h').unsafeGet().length);
  });
});

describe('sql query from candles', () => {
  it('should search candles for 1h', () => {
    expect(
      sql({
        amountAsset: '111',
        priceAsset: '222',
        params: {
          timeStart: new Date('2017-04-03T00:00:00.000Z'),
          timeEnd: new Date('2017-04-03T23:59:59.999Z'),
          interval: '1h',
        },
      })
    ).toMatchSnapshot();
  });

  it('should search candles for 1d', () => {
    expect(
      sql({
        amountAsset: '111',
        priceAsset: '222',
        params: {
          timeStart: new Date('2017-04-03T00:00:00.000Z'),
          timeEnd: new Date('2017-04-03T23:59:59.999Z'),
          interval: '1d',
        },
      })
    ).toMatchSnapshot();
  });

  it('should search candles for 1m', () => {
    expect(
      sql({
        amountAsset: '111',
        priceAsset: '222',
        params: {
          timeStart: new Date('2017-04-03T00:00:00.000Z'),
          timeEnd: new Date('2017-04-03T23:59:59.999Z'),
          interval: '1m',
        },
      })
    ).toMatchSnapshot();
  });
});
