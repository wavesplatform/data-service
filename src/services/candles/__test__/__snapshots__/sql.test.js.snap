// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sql query from candles should search candles for 1d 1`] = `"select \\"time_start\\", \\"amount_asset_id\\", \\"price_asset_id\\", \\"low\\", \\"high\\", \\"volume\\", \\"quote_volume\\", \\"max_height\\", \\"txs_count\\", \\"weighted_average_price\\", \\"open\\", \\"close\\", \\"interval_in_secs\\", \\"a_dec\\".\\"decimals\\" as \\"a_dec\\", \\"p_dec\\".\\"decimals\\" as \\"p_dec\\" from \\"candles\\" inner join \\"asset_decimals\\" as \\"a_dec\\" on \\"amount_asset_id\\" = \\"a_dec\\".\\"asset_id\\" inner join \\"asset_decimals\\" as \\"p_dec\\" on \\"price_asset_id\\" = \\"p_dec\\".\\"asset_id\\" where \\"amount_asset_id\\" = '111' and \\"price_asset_id\\" = '222' and \\"time_start\\" >= '2017-04-03 03:00:00.000' and \\"time_start\\" <= '2017-04-04 02:59:59.999' and \\"interval_in_secs\\" = 86400 order by \\"time_start\\" asc"`;

exports[`sql query from candles should search candles for 1h 1`] = `"select \\"time_start\\", \\"amount_asset_id\\", \\"price_asset_id\\", \\"low\\", \\"high\\", \\"volume\\", \\"quote_volume\\", \\"max_height\\", \\"txs_count\\", \\"weighted_average_price\\", \\"open\\", \\"close\\", \\"interval_in_secs\\", \\"a_dec\\".\\"decimals\\" as \\"a_dec\\", \\"p_dec\\".\\"decimals\\" as \\"p_dec\\" from \\"candles\\" inner join \\"asset_decimals\\" as \\"a_dec\\" on \\"amount_asset_id\\" = \\"a_dec\\".\\"asset_id\\" inner join \\"asset_decimals\\" as \\"p_dec\\" on \\"price_asset_id\\" = \\"p_dec\\".\\"asset_id\\" where \\"amount_asset_id\\" = '111' and \\"price_asset_id\\" = '222' and \\"time_start\\" >= '2017-04-03 03:00:00.000' and \\"time_start\\" <= '2017-04-04 02:59:59.999' and \\"interval_in_secs\\" = 3600 order by \\"time_start\\" asc"`;

exports[`sql query from candles should search candles for 1m 1`] = `"select \\"time_start\\", \\"amount_asset_id\\", \\"price_asset_id\\", \\"low\\", \\"high\\", \\"volume\\", \\"quote_volume\\", \\"max_height\\", \\"txs_count\\", \\"weighted_average_price\\", \\"open\\", \\"close\\", \\"interval_in_secs\\", \\"a_dec\\".\\"decimals\\" as \\"a_dec\\", \\"p_dec\\".\\"decimals\\" as \\"p_dec\\" from \\"candles\\" inner join \\"asset_decimals\\" as \\"a_dec\\" on \\"amount_asset_id\\" = \\"a_dec\\".\\"asset_id\\" inner join \\"asset_decimals\\" as \\"p_dec\\" on \\"price_asset_id\\" = \\"p_dec\\".\\"asset_id\\" where \\"amount_asset_id\\" = '111' and \\"price_asset_id\\" = '222' and \\"time_start\\" >= '2017-04-03 03:00:00.000' and \\"time_start\\" <= '2017-04-04 02:59:59.999' and \\"interval_in_secs\\" = 60 order by \\"time_start\\" asc"`;
